### PR TITLE
frontend: add category specific fields

### PR DESCRIPTION
**Pull request by Synar**

### CODE REVIEW

File: backend/src/models/comment.rs [hunk 1]
```diff
@@ -147,7 +153,7 @@ impl AdminComment {
                 RETURNING *
             )
             SELECT i.id, i.entity_id, i.author, i.text, i.data, i.created_at, i.updated_at, i.moderated, i.version, 
-                display_name as entity_display_name, category_id 
+                display_name as entity_display_name, category_id as entity_category_id
```
Comment (ghost) [inline backend/src/models/comment.rs]: AS should be capital

File: backend/src/models/comment.rs [hunk 2]
```diff
@@ -187,7 +195,7 @@ impl AdminComment {
                 RETURNING *
             )
             SELECT i.id, i.entity_id, i.author, i.text, i.data, i.created_at, i.updated_at, i.moderated, i.version, 
-                e.display_name as entity_display_name, e.category_id 
+                e.display_name as entity_display_name, e.category_id as entity_category_id
```
Comment (ghost) [inline backend/src/models/comment.rs]: Same.

File: backend/src/models/comment.rs [hunk 3]
```diff
@@ -210,7 +218,7 @@ impl AdminComment {
             AdminComment,
             r#"
             SELECT c.id, c.entity_id, c.author, c.text, c.data, c.created_at, c.updated_at, c.moderated, c.version,
-                e.display_name as entity_display_name, e.category_id
+                e.display_name as entity_display_name, e.category_id as entity_category_id
```
Comment (ghost) [inline backend/src/models/comment.rs]: Same.

File: backend/src/models/comment.rs [hunk 4]
```diff
@@ -245,7 +253,7 @@ impl AdminComment {
...
```
Comment (ghost) [inline backend/src/models/comment.rs]: Same.

File: frontend/components/CommentsDisplayer.vue [hunk 5]
```diff
@@ -34,7 +34,8 @@
         />
 
         <ViewerCommonFormFields
-          :fields="props.commentFormFields"
+          :fields="props.commentFormFields
+            .filter(field => field.categories == null || field.categories.includes(props.entityCategoryId))"
```
Comment (ghost) [inline frontend/components/CommentsDisplayer.vue:38]: Should not it be a triple equal here ?

File: frontend/components/admin/families/EditForm.vue [hunk 6]
```diff
@@ -189,23 +200,36 @@
...
                   @update:model-value="onIndexableChange(field)"
                 />
               </span>
+              <span class="flex items-center gap-6">
+                <AdminInputSwitchField
+                  :id="'is_categories_restricted_' + index"
+                  :model-value="!(field.categories == null)"
+                  :label="'Restriction par catégories'"
+                  :variant="hasFieldAttributeBeenEdited(index, 'categories')"
+                  @update:model-value="(is_categories_restricted: boolean) => {
```
Comment (ghost) [inline frontend/components/admin/families/EditForm.vue:213]: Maybe you should create a method inside the code section, you've added too much logic for the template section.

File: frontend/pages/admin/families/[id]/entities.vue [hunk 7]
```diff
@@ -21,6 +21,7 @@
           :on-save-callback="onSave"
           kind-name="entité"
           kind="entity"
+          :categories="categories"
```
Comment (ghost) [inline frontend/pages/admin/families/[id]/entities.vue]: Place it with other `:` parameters (before the kind)

**Review by ghost — APPROVED** [review]
Overall, LGTM. Nothing to say on the logic part, various comments on the style.

File: backend/src/models/comment.rs (hunk 1 above)
Comment (Synar) [reply backend/src/models/comment.rs]: Sure

File: frontend/components/CommentsDisplayer.vue (hunk 5 above)
Comment (Synar) [reply frontend/components/CommentsDisplayer.vue:38]: No, the null/undefined value comes from a conversion of the None value in Rust by utoipa, and whether this gets converted to a null or undefined value feels like an implementation details to me that I do not want to depend on, so == is intended.

File: frontend/pages/admin/families/[id]/entities.vue (hunk 7 above)
Comment (Synar) [reply frontend/pages/admin/families/[id]/entities.vue]: I do not see the point, but sure if you prefer

**Review by ghost — CHANGES_REQUESTED** [review]
You didn't change regenerate the sqlx cache file. Go on the backend folder and run `cargo sqlx prepare`.

**Review by ghost — APPROVED** [review]

**Pull request merged**